release.sh

```diff
@@ -0,0 +1,660 @@
+    pkg_name=`$GREP '^PACKAGE = ' Makefile | sed 's|PACKAGE = ||'`
+    pkg_version=`$GREP '^VERSION = ' Makefile | sed 's|VERSION = ||'`
+    tar_name="$pkg_name-$pkg_version"
+    targz=$tar_name.tar.gz
+    tag_name=$(echo "v$pkg_version" | sed 's|.0$||')
+
+    # Now get the tarballs from github directly to compute their checksum
+    remote_targz_url=$remote_url_root/$tag_name/$targz
+    echo "downloading the tarbal from $remote_targz_url"
+    curl $remote_targz_url > $targz
+
+    # Obtain the top commit SHA which should be the version bump
+    # It should not have been tagged yet (the script will do it later)
+    local_top_commit_sha=`git  rev-list --max-count=1 HEAD`
+    if [ $? -ne 0 ]; then
+	echo "Error: unable to obtain the local top commit id."
+	cd $top_src
+	return 1
+    fi
+
+    # Check that the top commit looks like a version bump
+    git diff --unified=0 HEAD^ | $GREP -F $pkg_version >/dev/null 2>&1
+    if [ $? -ne 0 ]; then
+	# Wayland repos use  m4_define([wayland_major_version], [0])
+	git diff --unified=0 HEAD^ | $GREP -E "(major|minor|micro)_version" >/dev/null 2>&1
+	if [ $? -ne 0 ]; then
+	    echo "Error: the local top commit does not look like a version bump."
+	    echo "       the diff does not contain the string \"$pkg_version\"."
+	    local_top_commit_descr=`git log --oneline --max-count=1 $local_top_commit_sha`
+	    echo "       the local top commit is: \"$local_top_commit_descr\""
+	    cd $top_src
+	    return 1
+	fi
+    fi
+
+    # Check that the top commit has been pushed to remote
+    remote_top_commit_sha=`git  rev-list --max-count=1 $remote_name/$remote_branch`
+    if [ $? -ne 0 ]; then
+	echo "Error: unable to obtain top commit from the remote repository."
+	cd $top_src
+	return 1
+    fi
+    if [ x"$remote_top_commit_sha" != x"$local_top_commit_sha" ]; then
+	echo "Error: the local top commit has not been pushed to the remote."
+	local_top_commit_descr=`git log --oneline --max-count=1 $local_top_commit_sha`
+	echo "       the local top commit is: \"$local_top_commit_descr\""
+	cd $top_src
+	return 1
+    fi
+
+    # If a tag exists with the the tar name, ensure it is tagging the top commit
+    # It may happen if the version set in configure.ac has been previously released
+    tagged_commit_sha=`git  rev-list --max-count=1 $tag_name 2>/dev/null`
+    if [ $? -eq 0 ]; then
+	# Check if the tag is pointing to the top commit
+	if [ x"$tagged_commit_sha" != x"$remote_top_commit_sha" ]; then
+	    echo "Error: the \"$tag_name\" already exists."
+	    echo "       this tag is not tagging the top commit."
+	    remote_top_commit_descr=`git log --oneline --max-count=1 $remote_top_commit_sha`
+	    echo "       the top commit is: \"$remote_top_commit_descr\""
+	    local_tag_commit_descr=`git log --oneline --max-count=1 $tagged_commit_sha`
+	    echo "       tag \"$tag_name\" is tagging some other commit: \"$local_tag_commit_descr\""
+	    cd $top_src
+	    return 1
```

Comment:
side-note: I'd like us to check for signed tags here but that's something better added to the xorg script